src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2186,5 +2186,182 @@ private async Task ValidateDeleteOrganizationAsync(Organization organization)
                 throw new BadRequestException("You cannot delete an Organization that is using Key Connector.");
             }
         }
+
+        public async Task DisableUserAsync(OrganizationUser organizationUser, Guid? disablingUserId)
+        {
+            if (organizationUser.Status == OrganizationUserStatusType.Disabled)
+            {
+                throw new BadRequestException("Already disabled.");
+            }
+
+            if (disablingUserId.HasValue && organizationUser.UserId == disablingUserId.Value)
+            {
+                throw new BadRequestException("You cannot disable yourself.");
+            }
+
+            if (organizationUser.Type == OrganizationUserType.Owner && disablingUserId.HasValue &&
+                !await _currentContext.OrganizationOwner(organizationUser.OrganizationId))
+            {
+                throw new BadRequestException("Only owners can disable other owners.");
+            }
+
+            if (!await HasConfirmedOwnersExceptAsync(organizationUser.OrganizationId, new[] { organizationUser.Id }))
+            {
+                throw new BadRequestException("Organization must have at least one confirmed owner.");
+            }
+
+            await _organizationUserRepository.DisableAsync(organizationUser.Id);
```

Comment:
Do we need to update the org user record in memory here to reflect the proper state after the sproc call?

Reply:
Yes, good catch, same on the bulk entries as well since they get returned in a collection of tuples.